### ISSUE TITLE
mapAccum and mapAccumM should be strict in their state

### ISSUE THREAD

**Issue by ndmitchell** [snoyberg/conduit#218]
That seems reasonable to expect, given that `foldM` works that way.

**Comment by snoyberg** [comment]
Agreed. Can you send a PR? I'm not at my computer now.

**Comment by snoyberg** [comment]
Got to a computer faster than I thought I would. If you want me to add this, no problem, just let me know.

**Comment by ndmitchell** [comment]
I'll have a pull request with you in a few mins.

**Comment by ndmitchell** [comment]
https://github.com/snoyberg/conduit/pull/219

**Comment by snoyberg** [comment]
Thanks. Uploaded as 1.2.5.

**Comment by ndmitchell** [comment]
I made this change because it seemed like the right thing to do. I've just spent two hours tracking down a Hoogle space leak that was only exhibiting on some machines (specifically the real Hoogle server) and not others (my development machine). Turned out to be entirely down to a not-strict accumulator which due to the old version of conduit was causing issues. Just shows that accumulators should definitely be strict!
